pkg/vsphere/tasks/waiter.go

```diff
@@ -70,10 +75,18 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 		}
 
 		log.Errorf("task failed: %s", err)
-		if !isTaskInProgress(err) {
+		if !needsRetry(target, err) {
```

Comment:
`needsRetry` calls `needsFix`. Maybe not have `needsRetry` call `needsFix` and instead call them separately here, caching the results for later reuse?